### ISSUE TITLE
Terminal frozen every time try to quit editor when LSP is launching

### ISSUE THREAD

**Issue by anzhi0708** [helix-editor/helix#2059]
### Summary

Add this line to ```~/.config/helix/config.toml```:
```
[editor.lsp]
display-messages = true
```
Then edit any source file with LSP support enabled, you see language server's messages (e.g. "fetching...", "25%", "3/4" and a circle-like animation) shown on the left conner. 

Once you see the messages (obviously it means your language server is starting), immediately type ```:wq``` and your terminal freezes until the language server is completely booted up. 

Try this on an old machine. It takes even longer for the language server to boot up and you can easily get a frozen terminal.

### Reproduction Steps

See above.

### Helix log

See #2003.

### Platform

macOS, Termux

### Terminal Emulator

iTerm2, Termux

### Helix Version

helix 22.03 (d4e45fd4)

**Comment by archseer** [comment]
This is because we are trying to wait for the formatter, `:q` is instant.

See also https://github.com/helix-editor/helix/issues/1639

**Comment by hazeycode** [comment]
> This is because we are trying to wait for the formatter, `:q` is instant.
> 
> See also #1639

In my opinion, waiting for the formatter is not a good idea. If the language server hangs, formatting should be abandoned at the earliest opportunity. I don’t care about formatting 95% of the time I want to compile source code I’m editing, like, immediately.

also see #2570

**Comment by pppKin** [comment]
Maybe we should have a timeout here? Like if the LS didn't respond in 200ms, write the file anyway.

**Comment by archseer** [comment]
> If the language server hangs, formatting should be abandoned at the earliest opportunity.

Agreed, I'd accept a patch that implements this. Might be easier to do once https://github.com/helix-editor/helix/pull/2267 is merged

**Comment by hhhapz** [comment]
To add to this, `:cq` should also never wait for the LS. It should instantly abort all actions, instead right now it hangs until the LS responds.
